x-pack/plugins/maps/public/connected_components/mb_map/draw_control/draw_feature_control/draw_feature_control.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Component } from 'react';
+import { Map as MbMap } from 'mapbox-gl';
+// @ts-expect-error
+import MapboxDraw from '@mapbox/mapbox-gl-draw';
+import { Feature } from 'geojson';
+import { i18n } from '@kbn/i18n';
+// @ts-expect-error
+import * as jsts from 'jsts';
+import { getToasts } from '../../../../kibana_services';
+import { DrawControl } from '../';
+import { DRAW_MODE, DRAW_TYPE } from '../../../../../common';
+import { DrawState } from '../../../../../common/descriptor_types';
+
+const geoJSONReader = new jsts.io.GeoJSONReader();
+
+export interface Props {
+  addNewFeatureToIndex: (indexName: string, geometry: unknown, path: string) => void;
+  disableDrawState: () => void;
+  removeFeatures: (featureIds: string[]) => void;
+  drawType: DRAW_TYPE;
+  drawState: DrawState;
+  drawMode: DRAW_MODE;
+  mbMap: MbMap;
+}
+
+export class DrawFeatureControl extends Component<Props, {}> {
+  _onFeaturesSelected = (mbDrawControl: MapboxDraw) => (e: { features: Feature[] }) => {
+    if (this.props.drawType === DRAW_TYPE.TRASH) {
+      const featureIds = e.features
+        .map((feature: Feature) => {
+          return feature.id ? `${feature.id}` : '';
+        })
+        .filter((id) => id);
+      this.props.removeFeatures(featureIds);
+      mbDrawControl.trash();
+    }
+  };
+
+  _onDraw = (mbDrawControl: MapboxDraw) => async (e: { features: Feature[] }) => {
```

Comment:
in practice, this will only be a single feature, so I'd remove the array-signature.

Reply:
This is a callback with signature dictated at the lib level. It will always be an array even if we only need a single feature out of it